src/auth/auth.service.ts

```diff
@@ -0,0 +1,132 @@
+import { userService } from "@/user/user.service";
+import { jwtTokenService } from "@/services/JwtTokenService";
+import {
+  UnauthenticatedError,
+  UnprocessableEntityError,
+} from "@/errors/http";
+import {
+  LoginForm,
+  loginFormSchema,
+  RegistrationForm,
+  registrationFormSchema,
+} from "./auth.forms";
+import { validateExistingUser, validateUserPassword } from "./auth.validators";
+import bcrypt from "bcryptjs";
+import { prisma } from "@/db/prisma";
+
+export class AuthService {
+  async register(data: RegistrationForm) {
+    const result = registrationFormSchema.safeParse(data);
+
+    if (!result.success) {
+      throw new UnprocessableEntityError({
+        errors: result.error.flatten().fieldErrors,
+      });
+    }
+
+    const { email, password, name } = data;
+
+    await validateExistingUser(email);
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const newUser = await userService.createUser(name, email, hashedPassword);
+    const tokenPair = jwtTokenService.generateTokenPair({ userId: newUser.id });
+
+    await prisma.refreshToken.create({
+      data: {
+        user_id: newUser.id,
+        token: tokenPair.refreshToken,
+        expires_at: jwtTokenService.getRefreshTokenExpirationDate(),
+      },
+    });
+
+    return {
```

Comment:
Please, consider using a domain model or just a DTO to send data from service layer back to client (controller in this case). This will help you to define a return type easily, make data consistent between layers and decrease a chance of not obvious issue in the future.
E.g. [here](https://github.com/Code-the-Dream-School/ii-practicum-team-2-back/blob/main/src/goal-type/goal-type.service.ts#L88).